File: Models/GoodRecipt/GrImportDocument.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS_WEBAPI.Models
{
    [Table("RII_GR_ImportDocument")]
    public class GrImportDocument : BaseEntity
    {

        [Required, ForeignKey(nameof(GrHeader))]
        public long HeaderId { get; set; }
        public virtual GrHeader Header { get; set; } = null!;

        [Required]
        public byte[] Base64 { get; set; } = null!;

    }
}

File: WebApi/Models/GrImportDocument.ts
import { BaseEntity } from './BaseEntity';
import { GrHeader } from './GrHeader';

export interface GrImportDocument extends BaseEntity {
  headerId: number;
  header?: GrHeader;
  
  // Base64 encoded document data
  base64: string; // In TypeScript, we'll use string for base64 data
}